Gems/ROS2/Code/Source/Odometry/ROS2OdometryCovariance.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ROS2OdometryCovariance.h"
+#include "AzCore/RTTI/RTTIMacros.h"
+#include "AzCore/base.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <array>
+#include <cstddef>
+
+namespace ROS2
+{
+    void ROS2OdometryCovariance::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<ROS2OdometryCovariance>()
+                ->Version(1)
+                ->Field("Linear covariance", &ROS2OdometryCovariance::m_linearCovariance)
+                ->Field("Angular covariance", &ROS2OdometryCovariance::m_angularCovariance);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<ROS2OdometryCovariance>("ROS2 Odometry Covariance", "Define Odometry covariance")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ROS2OdometryCovariance::m_linearCovariance,
+                        "Linear covariance",
+                        "Set ros linear covariance")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &ROS2OdometryCovariance::m_angularCovariance,
+                        "Angular covariance",
+                        "Set ros angular covariance");
+            }
+        }
+    }
+
+    std::array<double, 36> ROS2OdometryCovariance::GetRosCovariance() const
```

Comment:
Should use AZStd::array instead
```suggestion
    AZStd::array<double, 36> ROS2OdometryCovariance::GetRosCovariance() const
```

Reply:
Cannot use AZStd because ROS2 nav_msgs::msg::Odometry uses std::array.
Using AZStd would require one additional conversion.